File: src/backend/go/internal/models/file.go
package models

import "time"

// File is the GORM entity for uploaded-file metadata - analogous to an EF Core
// entity class. The on-disk `Location` is never exposed to clients directly;
// content is retrieved via the dedicated download endpoint instead.
type File struct {
	ID              uint       `gorm:"primarykey;autoIncrement"                json:"id"`
	Name            string     `gorm:"type:varchar(255);not null"              json:"name"`
	Extension       string     `gorm:"type:varchar(20);not null"               json:"extension"`
	Size            int64      `gorm:"not null"                                json:"size"`
	ContentType     *string    `gorm:"type:varchar(100);column:content_type"   json:"contentType"`
	Location        string     `gorm:"type:varchar(500);not null"              json:"-"`
	CreatedAt       time.Time  `gorm:"autoCreateTime"                          json:"createdAt"`
	CreatedByUserID *uint      `gorm:"index"                                   json:"createdByUserId"`
	UpdatedAt       *time.Time `gorm:"autoUpdateTime"                          json:"updatedAt"`
	UpdatedByUserID *uint      `gorm:"index"                                   json:"updatedByUserId"`
	CreatedByUser   *User      `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;foreignKey:CreatedByUserID" json:"-"`
	UpdatedByUser   *User      `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;foreignKey:UpdatedByUserID" json:"-"`
}

// TableName overrides the default table name - mirrors [Table("files")] in EF.
func (File) TableName() string {
	return "files"
}

File: src/backend/python/shared/models/file.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from shared.database import Base


class File(Base):
    """ORM model for an uploaded file (analogous to an EF Core entity)."""

    __tablename__ = "files"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    extension: Mapped[str] = mapped_column(String(20), nullable=False)
    size: Mapped[int] = mapped_column(BigInteger, nullable=False)
    content_type: Mapped[str | None] = mapped_column(String(100), nullable=True)
    location: Mapped[str] = mapped_column(String(500), nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), onupdate=func.now(), nullable=True)

    def __repr__(self) -> str:
        return f"<File id={self.id!r} name={self.name!r} size={self.size!r}>"
